test/block_labels_test.dart

```diff
@@ -0,0 +1,393 @@
+        Else([
+          out < 0,
+        ], label: elseLabel)
+      ]),
+    ], label: alwaysCombLabel);
+  }
+}
+
+class LabeledChainSequentialModule extends Module {
+  Logic reset;
+  Logic d;
+  Logic get q => output('q');
+  final clk = SimpleClockGenerator(10).clk;
+
+  String firstFfLabel;
+  String secondFfLabel;
+
+  LabeledChainSequentialModule(
+      this.reset, this.d, [this.firstFfLabel='ff_1',
+      this.secondFfLabel='ff_2']) {
+    reset = addInput('reset', reset);
+    d = addInput('d', d);
+    final q = addOutput('q');
+
+    final qInternal = LogicNet();
+    Sequential(
+      clk,
+      reset: reset,
+      resetValues: {
+        qInternal: 0
+      },
+      [
+        qInternal < d
+      ],
+      label: firstFfLabel
+    );
+    Sequential(
+      clk,
+      reset: reset,
+      resetValues: {
+        q: 0
+      },
+      [
+        q < qInternal
+      ],
+      label: secondFfLabel
+    );
+  }
+}
+
+class LabeledMultiBlockModule extends Module {
+  Logic a;
+  Logic b;
+  Logic c;
+  Logic d;
+
+  Logic get out_0 => output('out_0');
+  Logic get out_1 => output('out_1');
+
+  String firstBlockLabel;
+  String secondBlockLabel;
+
+  LabeledMultiBlockModule(
+      this.a, this.b, this.c, this.d, [this.firstBlockLabel='block_0',
+      this.secondBlockLabel='block_1']) {
+    a = addInput('a', a);
+    b = addInput('b', b);
+    c = addInput('c', c);
+    d = addInput('d', d);    
+    final out_0 = addOutput('out_0');
+    final out_1 = addOutput('out_1');
+
+    Combinational([out_0 < a & b], label: firstBlockLabel);
+    Combinational([out_1 < c ^ d], label: secondBlockLabel);
+  }
+}
+
+class LabeledSsaModule extends Module {
+  String firstBlockLabel;
+  String secondBlockLabel;
+
+  LabeledSsaModule(Logic a, [this.firstBlockLabel='block_0',
+      this.secondBlockLabel='block_1']) {
+    a = addInput('a', a, width: a.width);
+    final b = addOutput('b', width: a.width);
+    final c = addOutput('c', width: a.width);
+
+    final intermediate_0 = Logic(name: 'intermediate_0', width: a.width);
+    final intermediate_1 = Logic(name: 'intermediate_1', width: a.width);
+
+    final inc_0 = IncrModule(intermediate_0);
+    final inc_1 = IncrModule(intermediate_1);
+
+    Combinational.ssa((s) => [
+          s(intermediate_0) < a,
+          s(intermediate_0) < inc_0.result,
+          s(intermediate_0) < inc_0.result,
+        ], label: firstBlockLabel);
+
+    Combinational.ssa((s) => [
+          s(intermediate_1) < c,
+          s(intermediate_1) < inc_1.result,
+          s(intermediate_1) < inc_1.result,
+        ], label: secondBlockLabel);
+    b <= intermediate_0;
+    c <= intermediate_1;
+  }
+}
+
+class IncrModule extends Module {
+  Logic get result => output('result');
+  IncrModule(Logic toIncr) : super(name: 'incr') {
+    toIncr = addInput('toIncr', toIncr, width: toIncr.width);
+    addOutput('result', width: toIncr.width);
+    result <= toIncr + 1;
+  }
+}
+
+void main() {
```

Comment:
we should add some tests that actually generate SV and confirm some substring is inside, e.g.

```dart
final sv = mod.generateSynth();
expect(sv, contains('if(thing) : my_label');
```